### PR TITLE
Update resource and vocab markdown on lesson clone

### PR DESCRIPTION
**Pull request by bethanyaconnor**
Finishes [PLAT-1447](https://codedotorg.atlassian.net/browse/PLAT-1447).

When we clone scripts, we also clone resource and vocab. This means that the resource and vocab markdown becomes outdated.

We already find and replace this markdown when we serve this content so I just reused that logic here. I extended that functionality to be able to pass in a custom replace function that looks up the cloned resource/vocab and inserts that markdown there.

A few decisions I made:
- We can't assume the keys for resources and vocab always match across course version, so I created a map so we can find the new one
- It's possible for a lesson to reference a resource that isn't in that lesson. In that case, I just copied it to the destination course version.
- Because we need to know the cloned resource/vocab's new key, it's hard to do this same update to the script overview. I'm currently thinking, at least for now, that curriculum can spot check that field

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/models/vocabulary.rb [hunk 1]
```diff
@@ -125,6 +125,17 @@ def serialize_scripts
...
+  def copy_to_course_version(destination_course_version)
+    return self if course_version == destination_course_version
+    persisted_vocab = Vocabulary.where(word: word, course_version_id: destination_course_version.id).first
+    if persisted_vocab && !!persisted_vocab.common_sense_media == !!common_sense_media
+      persisted_vocab
+    else
+      copied_vocab = Vocabulary.create!(word: word, definition: definition, common_sense_media: common_sense_media, course_version_id: destination_course_version.id)
+      copied_vocab
+    end
+  end
```
Comment (davidsbailey) [inline dashboard/app/models/vocabulary.rb:137]: thank you for extracting this. I looked for other stuff to extract out of copy_to_unit but I didn't see anything that could be pulled out super cleanly.

**Review by davidsbailey — APPROVED** [review]
Looks great, Bethany! The proc is a nice choice for making this abstraction 👍

File: dashboard/app/models/vocabulary.rb (hunk 1 above)
Comment (bethanyaconnor) [reply dashboard/app/models/vocabulary.rb:137]: I had already extracted this same logic for resources so I'm not sure why I didn't do the same for vocab. 🤷‍♀️  Thanks for looking for more things to extract -- always happy to pull more things out if possible :)

**Comment by bethanyaconnor** [comment]
> Because we need to know the cloned resource/vocab's new key, it's hard to do this same update to the script overview. I'm currently thinking, at least for now, that curriculum can spot check that field

I realized I can do this the same way that this PR handles resources referenced but not in a lesson. I'm going to do that in a follow-up PR though as this one is already approved.

**Pull request merged**